File: Telerik/Reporting Q1 2014/Examples/CSharp/Html5Demo/Controllers/ReportsController.cs
namespace CSharp.Html5Demo.Controllers
{
    using System;
    using System.IO;
    using System.Linq;
    using System.Web;
    using Telerik.Reporting.Cache.Interfaces;
    using Telerik.Reporting.Services.Engine;
    using Telerik.Reporting.Services.WebApi;

    public class ReportsController : ReportsControllerBase
    {
        protected override IReportResolver CreateReportResolver()
        {
            var appPath = HttpContext.Current.Server.MapPath("~/");
            var reportsPath = Path.Combine(appPath, @"..\..\..\Report Designer\Examples");

            return new ReportFileResolver(reportsPath)
                .AddFallbackResolver(new ReportTypeResolver());
        }

        protected override ICache CreateCache()
        {
            return Telerik.Reporting.Services.Engine.CacheFactory.CreateFileCache();
        }
    }
}

File: Telerik/Reporting Q1 2014/Examples/VB/Html5Demo/Controllers/ReportsController.vb
Imports Telerik.Reporting.Cache.Interfaces
Imports Telerik.Reporting.Services.Engine
Imports Telerik.Reporting.Services.WebApi
Imports System.IO

Public Class ReportsController
    Inherits ReportsControllerBase

    Protected Overrides Function CreateReportResolver() As IReportResolver
        Dim appPath = HttpContext.Current.Server.MapPath("~/")
        Dim reportsPath = Path.Combine(appPath, "..\..\..\Report Designer\Examples")

        Return New ReportFileResolver(reportsPath).AddFallbackResolver(New ReportTypeResolver())
    End Function

    Protected Overrides Function CreateCache() As ICache
        Return Telerik.Reporting.Services.Engine.CacheFactory.CreateFileCache()
    End Function
End Class
